trace/trace.go

```diff
@@ -472,3 +580,39 @@ func (s *Span) createChildSpan(ctx context.Context, async bool) (context.Context
 	ctx = PutSpanInContext(ctx, newSpan)
 	return ctx, newSpan
 }
+
+type randomIdentifier struct {
+	randomBytes   func([]byte) (int, error)
+	incrementLock sync.Mutex
+	nonRandomId   uint64
+}
+
+// newRandomIdentifier returns a new randomIdentifier with the default pseudo-random
+// generator.
+func newRandomIdentifier() *randomIdentifier {
+	return &randomIdentifier{
+		randomBytes: rand.Read,
```

Comment:
maybe do an initial `rand.Uint64()` to initialize `nonRandomId`?  or do we really always want it to iterate starting at 0 (outside of tests)?